src/test/ui/const-generics/issue-61422.rs

```diff
@@ -0,0 +1,15 @@
+// run-pass
+
+#![feature(const_generics)]
+//~^ WARN the feature `const_generics` is incomplete and may cause the compiler to crash
+
+use std::mem;
+
+fn foo<const SIZE: usize>() {
+    let arr: [u8; SIZE] = unsafe {
+        let mut array: [u8; SIZE] = mem::uninitialized();
```

Comment:
We should think about how we handle this in terms of post monomorphization errors given that if you substitute `u8` for a larger type and set `SIZE` to a huge number then `foo` would error post monomorphization. Leave a comment for now? cc @oli-obk

Reply:
Isn't this an issue for large allocations in general? E.g. https://github.com/rust-lang/rust/issues/55878.

Reply:
Yes, and we can do the same with e.g.:
```rust
fn foo<T>() {
    let _x: [T; 10000000000000];
}

fn main() {
    foo::<[u8; 1000]>();
}
```
but that doesn't necessarily mean we should introduce another avenue for post monomorphization errors.

Reply:
Types can depend on constants and constants can depend on types - this is *not* a new avenue.

Reply:
@eddyb Even so, this increases the surface area for post monomorphization errors and in this case we can force the user to introduce a WF constraint `where [u8; N]:,`. We'll need to consider this at one time or another.

Reply:
I don't think this belongs in a test case for a random issue. If this is viewed as a separate concern, it should be opened in its own issue.

Reply:
Fine; I wanted this as a reminder when reviewing the test suite so I don't forget about it when we near stabilization... but I'll think about an issue instead.